tools/sched_ext/scx_serialise.bpf.c

```diff
@@ -86,134 +93,71 @@ struct {
 } task_ctx_map SEC(".maps");
 
 /*
- * Dispatch statistics. This map is used to keep track of the number of times
- * a main thread or a worker thread has been dispatched.
+ * The queue is used to manage the dispatch ordering of task groups. Each 
+ * task group represents a test case that originates from an executor identified by `eid`. 
+ * 
+ * `eid` are enqueued into this map using the `enqueue_eid_for_dispatch()` function when tasks are ready.
+ * `eid` are dequeued using `dequeue_eid_for_dispatch()` to dispatch tasks from specific groups.
+ *   in the group is dispatched.
  */
 struct {
-	__uint(type, BPF_MAP_TYPE_PERCPU_ARRAY);
-	__uint(key_size, sizeof(u32));
-	__uint(value_size, sizeof(u64));
-	__uint(max_entries, 2); /* [main, worker] */
-} stats SEC(".maps");
+	__uint(type, BPF_MAP_TYPE_QUEUE);
+	__uint(max_entries, 64);
+	__type(value, u32);
+} group_dispatch_queue SEC(".maps");
 
-static void stat_inc(u32 idx)
-{
-	u64 *cnt_p = bpf_map_lookup_elem(&stats, &idx);
-	if (cnt_p)
-		(*cnt_p)++;
+static void enqueue_eid_for_dispatch(u32 eid) {
+	int status = bpf_map_push_elem(&group_dispatch_queue, &eid, BPF_EXIST);
+	if (status < 0)
+        bpf_printk("[enqueue_eid_for_dispatch] Failed to update group_dispatch_queue\n");
+}
+
+static u32 dequeue_eid_for_dispatch() {
+	u32 eid;
+	if (bpf_map_pop_elem(&group_dispatch_queue, &eid) < 0) {
+		bpf_printk("[dequeue_eid_for_dispatch] Failed to dequeue group_dispatch_queue\n");
```

Comment:
Delete this printout -- any printing in `serialise_dispatch` is called too frequently -- it will hurt performance and also drowns out any other printing